openedx/features/course_experience/api/v1/views.py

```diff
@@ -47,13 +59,21 @@ def reset_course_deadlines(request):
         raise ParseError(_("Only 'course_key' is expected."))
 
     try:
-        reset_self_paced_schedule(request.user, course_key)
-
-        key = CourseKey.from_string(course_key)
-        if course_home_mfe_dates_tab_is_active(key):
-            body_link = get_microfrontend_url(course_key=course_key, view_name='dates')
+        course_key = CourseKey.from_string(course_key)
+        _course_masquerade, user = setup_masquerade(
+            request,
+            course_key,
+            has_access(request.user, 'staff', course_key)
+        )
+
+        missed_deadlines, missed_gated_content = dates_banner_should_display(course_key, user)
+        if missed_deadlines and not missed_gated_content:
```

Comment:
Used in the other version of this method. Doesn't look like it would hurt so I added it here.